webpack/redux/reducers/RedHatRepositories/enabled.js

```diff
@@ -1,28 +1,72 @@
+import Immutable from 'seamless-immutable';
+
 import {
   ENABLED_REPOSITORIES_REQUEST,
   ENABLED_REPOSITORIES_SUCCESS,
   ENABLED_REPOSITORIES_FAILURE,
+  REPOSITORY_ENABLED,
+  REPOSITORY_DISABLED,
 } from '../../consts';
 
-const initialState = { isLoading: true };
+const initialState = Immutable({ isLoading: true, repositories: [] });
+
+const flattenRepositorySets = sets =>
+  sets
+    .map(({
+      id: contentId, type, product: { id: productId }, repositories,
+    }) =>
+      repositories.map(repo => ({
+        type,
+        contentId,
+        productId,
+        ...repo,
+      })))
+    .reduce((a, b) => a.concat(b));
 
 export default (state = initialState, action) => {
   switch (action.type) {
+    case REPOSITORY_ENABLED:
+      return state.set('repositories', state.repositories.concat([action.repository]));
+
+    case REPOSITORY_DISABLED:
+      return state.set(
+        'repositories',
+        state.repositories.filter((repo) => {
+          const isMatch =
+            action.repository.arch === repo.arch &&
+            action.repository.contentId === repo.contentId &&
+            action.repository.productId === repo.productId &&
+            action.repository.releasever === repo.releasever;
+
+          return !isMatch;
+        }),
+      );
+
+    // const example = {
+    //   arch: 'x86_64',
+    //   contentId: '1952',
+    //   name: 'Red Hat Enterprise Linux 6 Server Kickstart x86_64 6.8',
+    //   productId: 5,
+    //   releasever: '6.8',
+    //   type: 'kickstart',
+    // };
+
     case ENABLED_REPOSITORIES_REQUEST:
-      return { ...state, ...{ isLoading: true } };
+      return state.set(['isLoading'], true);
 
     case ENABLED_REPOSITORIES_SUCCESS:
-      return {
-        ...state,
-        ...{
-          results: action.response.results,
-          isLoading: false,
-          success: true,
-        },
-      };
+      return Immutable({
+        repositories: flattenRepositorySets(action.response.results),
+        isLoading: false,
+        success: true,
```

Comment:
What is `success` used for?  Wouldn't we know whether it was success or failure based on the action?

Reply:
This should probably be changed to `error` with some information about the error. I wasn't sure what information we'll have or how we'll display it.